packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js

```diff
@@ -401,35 +401,73 @@ async function updateWalkthrough(context) {
-  ({ authConfig, resolverConfig } = await askAdditionalQuestions(context, parameters, authConfig, defaultAuthType, modelTypes));
-  await checkForCognitoUserPools(context, parameters, authConfig);
-  /* eslint-disable */
+  const updateOptionQuestion = {
+    type: 'list',
+    name: 'updateOption',
+    message: 'Select from the options below',
+    choices: [
+      {
+        name: 'Walkthrough all configurations',
+        value: 'all',
+      },
+      {
+        name: 'Update auth settings',
+        value: 'authUpdate',
+      },
+      {
+        name: 'Enable DataStore for entire API',
+        value: 'enableDatastore',
+      },
+    ],
+  };
 
-  const amplifyMetaFilePath = context.amplify.pathManager.getAmplifyMetaFilePath();
-  const amplifyMeta = context.amplify.readJsonFile(amplifyMetaFilePath);
+  let { updateOption } = await inquirer.prompt([updateOptionQuestion]);
 
-  if (amplifyMeta[category][resourceName].output.securityType) {
-    delete amplifyMeta[category][resourceName].output.securityType;
+  if (updateOption === 'enableDatastore') {
+    resolverConfig = {
+      project: { ConflictHandler: 'AUTOMERGE', ConflictDetection: 'VERSION' },
+    };
+  } else if (updateOption === 'authUpdate') {
+    /* eslint-disable */
+    ({ authConfig, defaultAuthType } = await askDefaultAuthQuestion(context, parameters));
+    authConfig = await askAdditionalAuthQuestions(context, parameters, authConfig, defaultAuthType);
+    await checkForCognitoUserPools(context, parameters, authConfig);
+    /* eslint-disable */
+  } else if (updateOption === 'all') {
+    /* eslint-disable */
+    ({ authConfig, defaultAuthType } = await askDefaultAuthQuestion(context, parameters));
+    ({ authConfig, resolverConfig } = await askAdditionalQuestions(context, parameters, authConfig, defaultAuthType, modelTypes));
+    await checkForCognitoUserPools(context, parameters, authConfig);
+    /* eslint-disable */
   }
 
-  amplifyMeta[category][resourceName].output.authConfig = authConfig;
-  let jsonString = JSON.stringify(amplifyMeta, null, '\t');
-  fs.writeFileSync(amplifyMetaFilePath, jsonString, 'utf8');
+  if (authConfig) {
+    const amplifyMetaFilePath = context.amplify.pathManager.getAmplifyMetaFilePath();
+    const amplifyMeta = context.amplify.readJsonFile(amplifyMetaFilePath);
 
-  const backendConfigFilePath = context.amplify.pathManager.getBackendConfigFilePath();
-  const backendConfig = context.amplify.readJsonFile(backendConfigFilePath);
+    if (amplifyMeta[category][resourceName].output.securityType) {
+      delete amplifyMeta[category][resourceName].output.securityType;
+    }
 
-  if (backendConfig[category][resourceName].output.securityType) {
-    delete backendConfig[category][resourceName].output.securityType;
-  }
+    amplifyMeta[category][resourceName].output.authConfig = authConfig;
+    let jsonString = JSON.stringify(amplifyMeta, null, '\t');
```

Comment:
Are we using tabs in all the places? We should stick to one, and I see in most of places, we use 4 spaces

Reply:
Done.